### PR TITLE
[SCB-2089] 支持服务中心rbac认证

### PR DESCRIPTION
**Pull request by jeho0815**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: dependencies/bom/pom.xml [hunk 1]
```diff
@@ -174,6 +174,11 @@
         <artifactId>foundation-vertx</artifactId>
         <version>${project.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.servicecomb</groupId>
+        <artifactId>foundation-auth</artifactId>
```
Comment (liubao68) [inline dependencies/bom/pom.xml:179]: delete this

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import com.netflix.config.DynamicPropertyFactory;
+
+public class TokenAuthHeaderProvider implements AuthHeaderProvider {
+  public static final String ACCOUNT_NAME_KEY = "servicecomb.credentials.account.name";
+
+  public static final String PASSWORD_KEY = "servicecomb.credentials.account.password";
+
+  public static final String CIPHER_KEY = "servicecomb.credentials.cipher";
+
+  private String registryName;
```
Comment (liubao68) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java:40]: registryName need support configuration

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    this.password = DynamicPropertyFactory.getInstance()
+        .getStringProperty(PASSWORD_KEY, null).get();
+    this.cipherName = DynamicPropertyFactory.getInstance()
+        .getStringProperty(CIPHER_KEY, DefaultCipher.DEFAULT_CYPHER).get();
+    if (StringUtils.isNotEmpty(accountName)) {
+      TokenCacheManager.getInstance().addTokenCache(registryName, accountName, password, getCipher());
+    }
+  }
+
+  public TokenAuthHeaderProvider(String registryName, String accountName, String password, String cipherName) {
```
Comment (liubao68) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java:61]: This constructor never used, can be deleted

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenCacheManager.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        tokenCacheWorker.schedule( // retry after 1 second
+            this::refreshToken, 1, TimeUnit.SECONDS);
+        return;
+      }
+      RbacTokenRequest request = new RbacTokenRequest();
+      request.setAccountName(new String(cipher.decrypt(accountName.toCharArray())));
+      request.setPassword(new String(cipher.decrypt(password.toCharArray())));
+      RbacTokenResponse rbacTokenResponse = serviceRegistryClient.getRbacToken(request);
+      LOGGER.info("refresh token successfully {}", rbacTokenResponse.getStatusCode());
+      if (StringUtils.isEmpty(this.token) && Status.UNAUTHORIZED.getStatusCode() == rbacTokenResponse.getStatusCode()) {
```
Comment (liubao68) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenCacheManager.java:183]: should not retry when using old version of service center

File: dependencies/bom/pom.xml (hunk 1 above)
Comment (jeho0815) [reply dependencies/bom/pom.xml]: done

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java (hunk 2 above)
Comment (jeho0815) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java:40]: the default constructor use the default name, if somebody want to use multi servicecenter, the can use TokenAuthHeaderProvider(String registryName, String accountName, String password, String cipherName)

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java (hunk 3 above)
Comment (jeho0815) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenAuthHeaderProvider.java:61]: this is used to multi service center scence

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenCacheManager.java (hunk 4 above)
Comment (jeho0815) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/auth/TokenCacheManager.java]: done

**Pull request merged**